dockerrotate/main.py

```diff
@@ -1,71 +1,78 @@
 """
-Free up space by rotating out old Docker images.
+Free up space by rotating out old Docker images and containers.
 """
-from argparse import ArgumentParser
-from datetime import timedelta
-import re
+from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter
 
-from dateutil import parser
 from docker import Client
-from docker.errors import APIError
 from docker.errors import NotFound
 from docker.utils import kwargs_from_env
 
+from dockerrotate.images import clean_images
+from dockerrotate.containers import clean_containers
 
-UNIX_SOC_ARGS = {"base_url": 'unix://var/run/docker.sock'}
-
-TEN_SECONDS = timedelta(seconds=10)
+UNIX_SOC_ARGS = {"base_url": "unix://var/run/docker.sock"}
 
 
 def parse_args():
-    parser = ArgumentParser()
+    parser = ArgumentParser(formatter_class=ArgumentDefaultsHelpFormatter)
     parser.add_argument(
-        "--use-env",
-        "-e",
+        "-e", "--use-env",
         action="store_true",
         help="Load docker connection information from standard environment variables.",
     )
     parser.add_argument(
-        "--clean-containers",
-        "-c",
+        "--dry-run",
         action="store_true",
-        help="Clean out old containers",
+        help="Do not remove anything",
     )
     parser.add_argument(
-        "--clean-images",
-        "-i",
-        action="store_true",
+        "--client-version",
+        help="Specify client version to use.",
+    )
+
+    subparsers = parser.add_subparsers()
+
```

Comment:
cleaning images and containers is now done through two separate calls to `docker-rotate`.
on the hosts where I run `docker-rotate` I make sure to clean containers before cleaning images.